OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -314,6 +305,19 @@ public void Nudge(Actor self, Actor nudger, bool force)
 			}
 		}
 
+		public bool IsLeaving()
+		{
+			if (CurrentMovementTypes.HasFlag(MovementType.Horizontal))
+				return true;
+
+			if (CurrentMovementTypes.HasFlag(MovementType.Turn))
+			{
+				return Path != null && Path.Count > 0;
```

Comment:
It may be possible to infer this without using `Path` by looking at `fromCell` and `toCell` or by setting a flag from `EnteringCell` / `FinishedMoving`.